metrics/src/main/java/com/expedia/adaptivealerting/metrics/functions/service/MetricQueryService.java

```diff
@@ -0,0 +1,98 @@
+package com.expedia.adaptivealerting.metrics.functions.service;
+
+import com.expedia.adaptivealerting.anomdetect.util.HttpClientWrapper;
+import com.expedia.adaptivealerting.metrics.functions.source.MetricFunctionsSpec;
+import com.expedia.adaptivealerting.metrics.functions.source.graphite.GraphiteQueryResult;
+import com.expedia.adaptivealerting.metrics.functions.source.graphite.ConstructSourceURI;
+import com.expedia.metrics.MetricData;
+import com.expedia.metrics.MetricDefinition;
+import com.expedia.metrics.TagCollection;
+import com.typesafe.config.Config;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.http.client.fluent.Content;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+
+@Slf4j
+public class MetricQueryService {
+    private HttpClientWrapper metricFunctionHttpClient;
+    private final String EMPTY_RESULT_FROM_SOURCE = "{}";
+    private final String METRICDATA_KEY_FOR_EMPTRY_DATA = "aggregator.producer.";
+    private final double METRICDATA_DEFAULT_VALUE = 0.0;
+    private final String METRIC_SOURCE_KEY = "metric-source";
+    /* For now source supported is graphite alone , non-graphite is just
+    *  a place holder for the common method
+    */
```

Comment:
I would suggest removing the non-Graphite stuff altogether. If we were to add support for other backends, I'd probably expect this to happen via SPI plugins instead of just building out a single class like `MetricQueryService` (see https://en.wikipedia.org/wiki/Open%E2%80%93closed_principle). In any event we aren't currently doing this so it seems better just to keep things tight and remove placeholders for future ideas. Keeping things tight is itself a way to make things easier to extend and/or discard.